crates/sargon/src/system/sargon_os/profile_state_holder.rs

```diff
@@ -94,10 +94,9 @@ impl ProfileStateHolder {
     where
         F: Fn(&Profile) -> T,
     {
-        let guard = self
-            .profile_state
-            .try_read()
-            .expect("Implementing hosts should not read and write Profile from multiple threads.");
+        let guard = self.profile_state.read().expect(
```

Comment:
We consider any poisoning of the lock as bad state, it is possible to read the poisoned state and possibly use it, but the state can be in an intermediary bad form.  So if any profile mutation panicked on a separate thread, any further operation on the state will be aborted.